iam.tf

```diff
@@ -285,28 +286,29 @@ data "aws_iam_policy_document" "kms_permissions" {
       "kms:GenerateDataKey*",
       "kms:DescribeKey"
     ]
-    resources = [element(concat(aws_kms_key.this.*.arn, [""]), 0)]
+    resources = [aws_kms_key.this[0].arn]
   }
 }
 
 resource "aws_iam_role_policy" "kms" {
   count  = var.iam_role_enabled && local.kms_enabled ? 1 : 0
   name   = "kms-permissions"
   role   = element(concat(aws_iam_role.this.*.name, [""]), 0)
-  policy = element(concat(data.aws_iam_policy_document.kms_permissions.*.json, [""]), 0)
+  policy = data.aws_iam_policy_document.kms_permissions[0].json
 }
 
 resource "aws_iam_user_policy" "kms" {
   count  = var.iam_user_enabled && local.kms_enabled ? 1 : 0
   name   = "kms-permissions"
   user   = concat(aws_iam_user.this.*.name, [""])[0]
-  policy = element(concat(data.aws_iam_policy_document.kms_permissions.*.json, [""]), 0)
+  policy = data.aws_iam_policy_document.kms_permissions[0].json
 }
 
 data "aws_iam_policy_document" "kms_key" {
   count = local.kms_enabled ? 1 : 0
 
   statement {
+    sid = "Enable IAM User Permissions"
```

Comment:
AFAIK, SID could not have spaces in it.

Ref: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_sid.html

Reply:
But we have spaces in current setting, should we consider removing space?